modules/core/src/main/scala/gem/Location.scala

```diff
@@ -10,46 +11,167 @@ import scalaz.Ordering.{EQ, GT, LT}
+      case m: Location.Middle => m
+      case _                  => sys.error("Location arguments do not form a Middle position: " + fi.toList.mkString(", "))
+    }
 
-  def fromList(is: List[Int]): Location =
-    new Location(trim(is)) {}
 
-  def fromIList(is: IList[Int]): Location =
-    new Location(trim(is)) {}
+  // Utility
 
-  def fromNel(n: NonEmptyList[Int]): Location =
-    new Location(trim(n.list)) {}
+  /** Finds `count` `Location`s that fall evenly distributed between `l0` and
+    * `l1`, assuming these locations are not the same.
+    *
+    * @param count how many locations to find
+    * @param start starting location (exclusive)
+    * @param end ending Location (exclusive)
+    *
+    * @return sorted list of `Location` where every element is GT l0 and LT l1
+    *         (or vice versa if l0 is GT l1)
+    */
+  def find(count: Int, start: Location, end: Location): IList[Location.Middle] = {
+    val Zero  = BigInt(0)
+    val One   = BigInt(1)
+    val Max   = BigInt(Int.MaxValue)
+    val Min   = BigInt(Int.MinValue)
+    val Radix = Max + Min.abs + One
 
-  implicit val OrderLocation: Order[Location] = Order.order { (l0, l1) =>
-    l0.toList
-      .zipAll(l1.toList, Int.MinValue, Int.MinValue)
-      .find { case (a, b) => a =/= b }
-      .fold[Ordering](EQ) { case (a, b) => (a < b) ?[Ordering] LT | GT }
+    def toBase10(loc: Location, len: Int): BigInt =
+      loc.positions.take(len).foldRight((Zero, One)) { case (i, (acc, pow)) =>
+        (acc + (BigInt(i) + Min.abs) * pow, pow * Radix)
+      }._1
+
+    def fromBase10(bi: BigInt): Location.Middle = {
+      @tailrec
+      def go(rem: BigInt, tail: IList[Int]): Location.Middle = {
+        val (a, b) = rem /% Radix
+        val head   = (b + Min).intValue
+        if (a === Zero) new Location.Middle(OneAnd(head, tail)) {} else go(a, head +: tail)
+      }
+      go(bi, IList.empty)
+    }
+
+    @tailrec
+    def go(len: Int): IList[Location.Middle] = {
+      val start10 = toBase10(start, len)
+      val end10   = toBase10(end, len)
+      val avail   = end10 - start10 - One
+
+      if (avail < count)
+        go(len + 1)
+      else {
+        val incr = (BigDecimal.exact(avail) / (count + 1)).setScale(0, BigDecimal.RoundingMode.CEILING).toBigInt
+        IList.fromList((1 to count).toList.map(i => fromBase10(start10 + (incr * i))))
+      }
+    }
+
+    if ((count <= 0) || (start >= end)) INil[Location.Middle]
+    else go(start.minPrefixLength max end.minPrefixLength)
   }
 
+  // Type Classes
+
+  implicit val OrderLocation: Order[Location] = Order.order(Function.untupled {
+    case (Beginning,  Beginning )                => EQ
+    case (End,        End       )                => EQ
+    case (Middle(m0), Middle(m1)) if (m0 === m1) => EQ
+    case (l0,         l1        )                =>
+      l0.positions.zip(l1.positions)
+        .find { case (a, b) => a =/= b }
+        .fold[Ordering](EQ) { case (a, b) => (a < b) ?[Ordering] LT | GT }
+  })
```

Comment:
I like how you're using a stream here so you don't have to consider the "unequal length" case.